File: Stack.java
import java.lang.*;
import java.util.*;

class node {
    public int data;
    public node next;
}

class Stackk {

    public node Head;
    public int Count;

    public Stackk() {
        Head = null;
        Count = 0;
    }

    public void Push(int no) {
        node newn = new node();

        newn.data = no;
        newn.next = null;

        if (Head == null) {
            Head = newn;
        } else {
            newn.next = Head;
            Head = newn;
        }

        Count++;
    }

    public void Display() {
        node temp = Head;
        System.out.print("Elements from Stack are:");
        while (temp != null) {
            System.out.print("|" + temp.data + "|->");
            temp = temp.next;
        }
        System.out.println("NULL");
    }

    public int CountNode() {
        return Count;
    }

    public void Pop() {
        if (Head == null) {
            return;
        }

        else {
            node temp = Head;
            int no = temp.data;
            Head = temp.next;
            temp = null;
            System.out.println("Removed element are" + no);

        }
        Count--;
    }

}

class Stack {
    public static void main(String args[]) {

        Stackk obj1 = new Stackk();

        obj1.Push(11);
        obj1.Push(21);
        obj1.Push(51);
        obj1.Push(101);

        obj1.Display();

        System.out.println("Count of elements are " + obj1.CountNode());

        obj1.Pop();
        obj1.Pop();

        obj1.Display();

        System.out.println("Count of elements are " + obj1.CountNode());

        obj1.Push(111);
        obj1.Display();

        obj1.Pop();
        obj1.Display();
    }
}

File: Stack.c
#include<stdio.h>
#include<stdlib.h>

struct node
{
    int data;
    struct node * next;

};


typedef struct node NODE;
typedef struct node * PNODE;
typedef struct node ** PPNODE;

void Push(PPNODE Head,int no)
{
    PNODE newn=NULL;
    newn=(PNODE)malloc(sizeof(NODE));
    PNODE temp=*Head;

    newn->data=no;
    newn->next=NULL;


    if(*Head==NULL)
    {
        *Head=newn;
    }
    else
    {
       newn->next=*Head;
       *Head=newn;
    }

}


void Pop(PPNODE Head)
{
    PNODE temp=*Head;
    int no=0;
    
    if(*Head==NULL)
    {
        return;
    }

    else
    {
        
        temp=*Head;
        no=temp->data;
        *Head=temp->next;
        free(temp);
        printf("Poped data is:%d\n",no);
        
    }
}


void Display(PNODE head)
{
    printf("Elemnts from Stack are \n");
    while(head!=NULL)
    {
        printf("|%d|->",head->data);
        head=head->next;
    }
    printf("NULL\n");
}

int Count(PNODE head)
{
    int iCnt=0;
    while(head!=NULL)
    {
        iCnt++;
        head=head->next;
    }

    return iCnt;
}



int main()
{
   PNODE First=NULL;
   int iRet=0;

   Push(&First,11);
   Push(&First,21);
   Push(&First,51);

   Display(First);

   
   iRet=Count(First);
   printf("Count is:%d",iRet);



   Pop(&First);

   Display(First);



    return 0;
}